### ISSUE TITLE
Out of range date in strftime() causes report failure  

### ISSUE THREAD

**Issue by toropok** [senaite/senaite.databox#32]
If user puts date with year value earlier than 1900 year in `date_from` and `date_to` field:

<img width="979" alt="Screenshot 2023-12-08 at 17 21 27" src="https://github.com/senaite/senaite.databox/assets/14867014/31c004b0-110b-492c-a3d0-d9f47294036a">

that causes exception in `databox_controls.pt`:

```
...
Module Products.PageTemplates.engine, line 378, in __call__
  Module z3c.pt.pagetemplate, line 176, in render
  Module chameleon.zpt.template, line 302, in render
  Module chameleon.template, line 215, in render
  Module chameleon.template, line 192, in render
  Module 2cfdf7d550b9a6fb1f96f131bf2f6928, line 993, in render
  Module zope.tales.pythonexpr, line 73, in __call__
   - __traceback_info__: (date_to and date_to.strftime('%Y-%m-%d'))
  Module <string>, line 1, in <module>
ValueError: year=23 is before 1900; the datetime strftime() methods require year >= 1900

 - Expression: "python:date_to and date_to.strftime('%Y-%m-%d')"
 - Filename:   ... rc/senaite/databox/browser/templates/databox_controls.pt
 - Location:   (line 164: col 43)
 - Source:     ... es="value python:date_to and date_to.strftime('%Y-%m-%d')"
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "provider:plone.abovecontentbody"
 - Filename:   ... te/core/browser/main_template/templates/main_template.pt
 - Location:   (line 132: col 84)
 - Source:     ... 
...
```

would you mind if we add year check in the setter method, like this:

```python
...
from senaite.core.api.dtime import to_DT
...
    # DATE TO

    def _set_date_to(self, value):
        self.context.date_to = to_DT(value) if to_DT(value).year() >= 1900 else None
```

Or there any better options?

**Comment by toropok** [comment]
If there any way to prevent user from the wrong input - thats great

**Comment by ramonski** [comment]
Hi @toropok,
we handle this error already here:
https://github.com/senaite/senaite.core/blob/2.x/src/senaite/core/api/dtime.py#L376
However, I don't know how the catalog search will be affected by dates before 1900

**Comment by toropok** [comment]
> Hi @toropok, we handle this error already here: https://github.com/senaite/senaite.core/blob/2.x/src/senaite/core/api/dtime.py#L376 However, I don't know how the catalog search will be affected by dates before 1900

@ramonski aha, cool. Does it make sense to remove strftime conversion from the [149](https://github.com/senaite/senaite.databox/blob/2d363b3c13e5e337974724620bc63fc392075596/src/senaite/databox/browser/templates/databox_controls.pt#L149) and [164](https://github.com/senaite/senaite.databox/blob/2d363b3c13e5e337974724620bc63fc392075596/src/senaite/databox/browser/templates/databox_controls.pt#L164C26-L164C26) lines at [`databox_controls.pt` ](https://github.com/senaite/senaite.databox/blob/2d363b3c13e5e337974724620bc63fc392075596/src/senaite/databox/browser/templates/databox_controls.pt#L149C1-L150C1)????

**Comment by ramonski** [comment]
The `value` is needed for this input field to work properly.
But you could try to do the conversion in the view and use the `dtime.date_to_string` function there or directly in the template, e.g. something like this might work:

```html
<input type="date"
       class="form-control"
       tal:define="date_from here/date_from"
       tal:attributes="value python:view.dtime.date_to_string(date_from) if date_from else ''"
       name="senaite.databox.date_from">
```

**Comment by toropok** [comment]
> ```
> date_to_string(
> ```

yeah, that works (if import dtime directly in DataBoxView class, otherwise it is invisible in `databox_controls.pt`):

```python
class DataBoxView(ListingView):
    """The default DataBox view
    """
    from senaite.core.api import dtime
    template = ViewPageTemplateFile("templates/databox_view.pt")
```

should we add it as PR?

**Comment by ramonski** [comment]
> should we add it as PR?

Yes, that would be great! Then it will get part of the upcoming 2.5 release.
Thanks!

**Comment by toropok** [comment]
> > should we add it as PR?
> 
> Yes, that would be great! Then it will get part of the upcoming 2.5 release. Thanks!

im very sorry I missed somehow your message :( my bad

created that micro PR: #33

**Comment by ramonski** [comment]
> im very sorry I missed somehow your message :( my bad

No worries, we can make another dot release if more comes
